lib/fizzy/execute.cpp

```diff
@@ -159,6 +159,22 @@ Instance instantiate(const Module& module)
     // NOTE: fill it with zeroes
     bytes memory(memory_min * page_size, 0);
 
+    // Fill out memory based on data segments
+    for (const auto& data : module.datasec)
+    {
+        uint64_t offset;
+        if (data.offset.kind == ConstantExpression::Kind::Constant)
+            offset = data.offset.value.constant;
+        else
+            throw std::runtime_error("data initialization by imported global is not supported yet");
+
+        // FIXME: assert in parser
+        assert(data.memidx == 0);
+        // NOTE: these instructions can overlap
+        assert((offset + data.init.size()) <= (memory_max * page_size));
+        __builtin_memcpy(memory.data() + offset, data.init.data(), data.init.size());
```

Comment:
I would vote for using `memcpy` to minimize non-standard code, if it doesn't make performance/correctness difference

Reply:
You mean C memcpy or C++ std::copy ?

Reply:
`std::copy_n` should be fine.

Reply:
BTW, the `__builtin_memcpy()` and `memcpy()` is the same, just you don't need to include `<cstring>` for the first one.